scripts/linters/pylint_extensions.py

```diff
@@ -2109,6 +2109,44 @@ def visit_call(self, node):
                     break
 
 
+class DisallowBlankLinesBelowFunctionDefinitionChecker(checkers.BaseChecker):
+    """Custom pylint checker to ensure there are no blank lines
+    below function definition.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+    name = 'disallow-blank-lines-below-function-definition'
+    priority = -1
+    msgs = {
+        'C0034': (
+            'Please remove the blank line(s) below the function definition.',
+            'remove-blank-lines-below-function-definiton.',
+            'Disallow blank lines below function definition.'
+        ),
+    }
+
+    def visit_functiondef(self, node):
+        """Visits every function and ensures there are no blank lines below
+        function definition.
+
+        Args:
+            node: astroid.scoped_nodes.Function. Node for a function or method
+                definition in the AST.
+        """
+        line_number = node.fromlineno
+        while True:
```

Comment:
Can you think about this once again, and check for a case where it can go into infinite loop?

Reply:
@sajalasati but if that is the case, this should never stop locally also. Locally, it is passing all the cases and backend tests are also passing locally. See here
![Screenshot from 2021-03-12 17-39-49](https://user-images.githubusercontent.com/56195849/110939569-a21cef80-835b-11eb-9678-7f7e151659e4.png)

I ran  python -m scripts.run_backend_tests for backend tests. lint tests are covered in this script right?

Reply:
@Mayank-gaur I am not referring to the backend tests you wrote, your code runs fine on them and hence it passes (and runs quickly) that.

Lint checks are taking infinite time to run, so it means you have probably missed a test case on which it will never stop running (which you have not written that in Test class also but occurs in some file of the codebase). Can you test your code on other cases too to find out where the code goes into infinite loop? Like some tests with missing docstring in function, or similar other variations. You can also run the linter on specific files (see [here](https://github.com/oppia/oppia/wiki/Lint-Checks#how-to-run-the-linter)) to see how the code performs on those files.

Is the problem clear to you now?

Reply:
Yes I got it @sajalasati . The lint operation after submitting the PR, which is failing  now is similar to 'python -m scripts.linters.pre_commit_linter --path /home/mayank/opensource/oppia' command locally, right? Is the path right?

Reply:
Also, some other checks are also failing in automatic evaluation, like Lighthouse accessiblity test,combined  backend test coverage and circleCI tests. I have to work on removing them also right, or are they because of this missed case only?

Reply:
> Yes I got it @sajalasati . The lint operation after submitting the PR, which is failing now is similar to 'python -m scripts.linters.pre_commit_linter --path /home/mayank/opensource/oppia' command locally, right? Is the path right?

Just give the path to any python file relative to your oppia root directory. eg `python -m scripts.linters.pre_commit_linter --path=core/domain/user_services.py`. Try for at least 4-5 files.

> Also, some other checks are also failing in automatic evaluation, like Lighthouse accessiblity test,combined backend test coverage and circleCI tests. I have to work on removing them also right, or are they because of this missed case only?

You should work on fixing Lint checks and Backend tests (others are not exactly related, some are flakes too). For backend tests see the log [here](https://github.com/oppia/oppia/pull/11980/checks?check_run_id=2090422880#step:20:302) - you need to make sure that backend tests cover each line of code that you write.

Reply:
okay @sajalasati

Reply:
I think lint checks are corrected now. I ran 'scripts.linters.pre_commit_linter --path /home/mayank/opensource/oppia' i.e ran the linter test on the whole oppia directory, and I got some lint errors which I have corrected now. 
![Screenshot from 2021-03-13 02-43-54](https://user-images.githubusercontent.com/56195849/110999516-7f63f880-83a6-11eb-8ed3-3acc8b183384.png)

I think infinite loop problem is solved now. I searched for the line starting with keyword def and then checked the next line status as blank or not.

Reply:
Please correct the lint errors in these files and push the changes for review once you have completed it.

Reply:
Yes @sajalasati I have already corrected them. But what should I do about the failing backend test?
![Screenshot from 2021-03-13 13-41-39](https://user-images.githubusercontent.com/56195849/111023956-0db99800-8402-11eb-97dc-2d265e4fdc6e.png)
It is showing 100% in all other cases but in pylint extensions, it is showing 99%. In the last column also, it is showing that only 1 cover is missing.  How should I proceed in resolving this?

Reply:
> It is showing 100% in all other cases but in pylint extensions, it is showing 99%. In the last column also, it is showing that only 1 cover is missing. How should I proceed in resolving this?

@Mayank-gaur Regarding this I already replied in the [above comment](https://github.com/oppia/oppia/pull/11980#discussion_r593282321) - you need to make sure that the tests you have written cover **every** line of code you have written. Probably with the current tests, the execution never reaches line 2142, so you should add a test that does that.

Reply:
Thanks @sajalasati for the help. But I am still not sure how to work towards it, like there is no traceback and I dont know where to look for the bug. Can you please give me some more detail on how to proceed on this?

Reply:
> Thanks @sajalasati for the help. But I am still not sure how to work towards it, like there is no traceback and I dont know where to look for the bug. Can you please give me some more detail on how to proceed on this?

**Gist of my reply**: Don't think about traceback and all. The way is to simply think about the case when it will go into the line 2142. Or the better ques you should ask yourself is why did you even write that line of code if you can't figure out when the execution will reach that line. Will your logic become incorrect or will miss you any case if you remove it? That's how I wanted you to think when you find coverage error in backend tests.

**Quick fix**: The line 2142 would be needed for one example case where the def line spans across multiple lines:
```python
def this_is_a_test_name_that_spans_across_multiple_lines(
        attribute_name_too_long_to_fit_in_single_line):
    pass
```
Here if you just break off at `def` line and check the next line that shouldn't be correct, right?

**Note:** As author of the PR it is your responsibility to ensure that the approach (logic) used is correct. You can probably fix (improve) your approach later on, but you should be able to explain why you have used certain approach currently.

Reply:
Yes @sajalasati    i understood the meaning of backend coverage tests a little while after, that they mean that execution is never reaching that line. I corrected that also, PTAL @sajalasati . This time i think that PR is ready to merge, but please guide me if any further changes are also required. 

I earlier thought that it would be good to search for def by incrementing line_number,like i did with the raw checker , but that really isn't required as visit function node always begins at first line.  I realised that later and corrected it, so all backemd tests are also passing now.
All this time could have been saved if i would have just discussed my approach before coding. I will ensure that my workflow of next PR is quite smooth.  Thanks!